x-pack/plugin/core/template-resources/src/main/resources/elastic-connectors.json

```diff
@@ -1,18 +1,16 @@
 {
-  "template": {
-    "aliases": {
-      ".elastic-connectors":  {}
-    },
-    "mappings": {
+  "settings": {
+    "index": {
+      "number_of_shards": "1",
+      "auto_expand_replicas": "0-1"
+    }
+  },
+  "mappings": {
+    "_doc": {
       "dynamic": "false",
```

Comment:
Consider setting `dynamic` to `strict` here.

Reply:
I think it's a good best practice. It safeguards against bugs if we add additional fields that should not be in the schema. If we decide to change the schema (add fields for example) there is a mechanism to version the system index - we do this with query rules.

Reply:
Changing the mappings to `strict` introduced a lot of problems, because some fields expect dynamic content (e.g. the `configuration` field for `.elastic-connectors`). 
I still think `strict` is good to have for the base-level fields. I've opted to add `dynamic: "false"` to any sub-fields that can potentially have dynamic content applied to them. You can see these in this commit: https://github.com/elastic/elasticsearch/pull/118991/commits/8da3e1c4b1186b2665f78ff0d7c203058fea3a3a
Does this sound like a good compromise?